### PR TITLE
[coordinator] Add FetchTagged request-level read consistency

### PR DESCRIPTION
**Pull request by rallen090**
**What this PR does / why we need it**:

Allows setting read consistency and iteration compare strategy at the FetchTagged request level.

**Special notes for your reviewer**:

**Does this PR introduce a user-facing and/or backwards incompatible change?**:

```release-note

```

**Does this PR require updating code package or user-facing documentation?**:

```documentation-note

```

### CODE REVIEW

**Review by ryanhall07 — APPROVED** [review]
nice!

File: src/query/api/v1/handler/prometheus/handleroptions/fetch_options.go [hunk 1]
```diff
@@ -333,6 +347,26 @@ func (b fetchOptionsBuilder) newFetchOptions(
...
+		"readConsistencyLevel")
+	if err != nil {
+		return nil, nil, err
+	}
+	if ok {
+		l := topology.ReadConsistencyLevel(v)
+		fetchOpts.ReadConsistencyLevel = &l
+	}
+
+	v, ok, err = TryParseValue(req, headers.IterateEqualTimestampStrategyHeader,
```
Comment (arnikola) [inline src/query/api/v1/handler/prometheus/handleroptions/fetch_options.go:360]: Should be better to utilize the existing string parsing defined for these: https://github.com/m3db/m3/blob/master/src/dbnode/encoding/iterators_types.go#L78; a little annoying to have to match the index of the mode we want when setting headers

Comment (arnikola) [reply src/query/api/v1/handler/prometheus/handleroptions/fetch_options.go:360]: Same for the read consistency level

Comment (arnikola) [reply src/query/api/v1/handler/prometheus/handleroptions/fetch_options.go:360]: Alternatively will also need some way of enforcing that `encoding.IterateEqualTimestampStrategy(v)` is a valid strategy

Comment (rallen090) [reply src/query/api/v1/handler/prometheus/handleroptions/fetch_options.go:360]: Done

**Pull request merged**